core/orchestrator.go

```diff
@@ -120,6 +120,8 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 		CreationRoundBlockHash: ethcommon.BytesToHash(payment.ExpirationParams.CreationRoundBlockHash),
 	}
 
+	var firstErr error
```

Comment:
Personally I would prefer not to use name return variables at all - they makes code harder to read.

Reply:
`firstErr` is actually removed in this [commit](https://github.com/livepeer/go-livepeer/pull/937/commits/d7dcbd3bddc86ddda37f269dfb0c5b26956abe6f) from a follow up PR. So, perhaps we can leave this as-is and continue any additional discussion on how the error is returned from `ProcessPayment()` based on the referenced commit in #937?

Reply:
Sure, didn't get to that one yet.

Reply:
As a general note on style, agreed with Ivan in that I prefer not to use named return variables since it's less explicit and requires holding more mental context for the function.

Given the emphasis that golang places on explicitness, I'm surprised this is even a feature in the language.

Reply:
Actually I find it to be more clear most times especially when using the go plugin for vscode that uses `godef` when browsing through code it actually helps as a form of documentation rather than having to go into the call chain to see what the context of the return value is. 

I'm pretty used to this in solidity as well. But perhaps a discussion to be had elsewhere 😄 .